hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/storage/NotificationDispatcher.java

```
@@ -62,4 +64,33 @@ public NotificationDispatcher(StorageAdapter storageAdapter, String feedId) {
                     }
                 });
     }
+
+    @Override
+    public <L> void receivedEvent(AvailEvent<L> event) {
+        MonitoredEndpoint<EndpointConfiguration> endpoint = event.getSamplingService().getMonitoredEndpoint();
+        String endpointTenantId = endpoint.getEndpointConfiguration().getTenantId();
+        String tenantId = (null != endpointTenantId) ? endpointTenantId
+                : storageAdapter.getStorageAdapterConfiguration().getTenantId();
+
+        event.getChanged().keySet().stream()
+                .filter(mi -> mi.getResource().getResourceType().getNotifications().contains(NotificationType.AVAIL_CHANGE))
+                .forEach(mi ->  {
+                    CanonicalPath cp = CanonicalPath.of()
+                            .tenant(tenantId)
+                            .feed(feedId)
+                            .resource(mi.getResource().getID().getIDString())
+                            .get();
+                    try {
+                        NotificationPayloadBuilder b = storageAdapter.createNotificationPayloadBuilder();
+                        b.addNotificationType(NotificationType.AVAIL_CHANGE);
+                        b.addProperty("resourceType", mi.getResource().getResourceType().getName().getNameString());
+                        b.addProperty("resourcePath", cp.toString());
+                        b.addProperty("availType", mi.getType().getName().getNameString());
+                        b.addProperty("newAvail", event.getChanged().get(mi).name());
+                        storageAdapter.store(b, 0);
+                    } catch (Exception e) {
+
```

should we log an error message here rather than silently ignore it?

---

Sure, not sure why it wasn't before.